__tests__/nametag.test.tsx

```diff
@@ -1,16 +1,33 @@
 import '@testing-library/jest-dom'
-import { render, screen } from '@testing-library/react'
+import { render, screen, fireEvent } from '@testing-library/react'
+import { userEvent } from '@testing-library/user-event'
+
 import Page from '../app/nametag/page'
 
 jest.mock('next/navigation', () => jest.requireActual('next-router-mock'))
 jest.mock('../lib/zoomapi', () => jest.requireActual('../lib/fakezoomapi'));
 
 describe('Page', () => {
-  it('renders a heading', () => {
+  it('renders the heading and input fields', () => {
     render(<Page />)
 
-    const heading = screen.getByRole('heading', { level: 1 })
-
-    expect(heading).toBeInTheDocument()
+    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
+    expect(screen.getByText('Full Name')).toBeInTheDocument();
+    expect(screen.getByText('Preferred Name')).toBeInTheDocument();
+    expect(screen.getAllByText('Select Pronouns')[0]).toBeInTheDocument();
+    expect(screen.getByText('Self Disclosure')).toBeInTheDocument();
   })
+
+  it('toggles the nametag display on switch change', async () => {
+    render(<Page />)
+
+    let switchInput = screen.getByTestId('colored-switch');
+    expect(switchInput).not.toBeChecked();
+
+    // fireEvent.change(switchInput, { target: { 'aria-checked': true } });
+    await userEvent.click(switchInput);
+
+    // need to revise this test case
```

Comment:
In this TODO comment, can we link to the notion task covering the bug in question?